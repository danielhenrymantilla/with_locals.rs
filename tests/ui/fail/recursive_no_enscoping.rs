include!("../prelude.rs");

struct Implementor;

impl Implementor {
    #[with('local, recursive = true)]
    fn foo (&self)
      -> &'local ()
    {
        &()
    }

    #[with('local, recursive = true)]
    fn bar (self: &'_ Self)
      -> &'local ()
    {
        &()
    }
}

trait Trait {
    #[with('local, recursive = true)]
    fn foo (&self)
      -> &'local ()
    {
        &()
    }

    #[with('local, recursive = true)]
    fn bar (self: &'_ Self)
      -> &'local ()
    {
        &()
    }
}
