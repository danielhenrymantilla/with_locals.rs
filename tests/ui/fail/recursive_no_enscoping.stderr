error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> tests/ui/fail/recursive_no_enscoping.rs:29:5
   |
29 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `with` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> tests/ui/fail/recursive_no_enscoping.rs:22:5
   |
22 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `with` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Missing `#[with]` on the enscoping `impl` or `trait` block
  --> tests/ui/fail/recursive_no_enscoping.rs:13:5
   |
13 |     #[with(recursive = true)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `with` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Missing `#[with]` on the enscoping `impl` or `trait` block
 --> tests/ui/fail/recursive_no_enscoping.rs:6:5
  |
6 |     #[with(recursive = true)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `with` (in Nightly builds, run with -Z macro-backtrace for more info)
